From 305bc215d739b88a2fb8ad4129c0303e9687decc Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Fri, 1 Dec 2023 16:09:00 +0100
Subject: [PATCH 0358/1695] drivers: stm32_tamp: add support for masking
 potential tamper reset

When "st,tamp-mask-pot-reset" property is set in the device tree TAMP
node, some resource aren't reset when a potential tamper even occurs.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git b9cb834bdb099c3ac2aad96e77264eaffe6f6ac0 drivers: stm32_tamp: add support for masking potential tamper reset]
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Change-Id: I093f053a3226297af6ed76ac92afcb42e022a728
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/343477
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-by: Maxime MERE <maxime.mere@foss.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
Tested-by: Maxime MERE <maxime.mere@foss.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
---
 core/drivers/stm32_tamp.c         | 6 ++++++
 core/include/drivers/stm32_tamp.h | 1 +
 2 files changed, 7 insertions(+)

diff --git a/core/drivers/stm32_tamp.c b/core/drivers/stm32_tamp.c
index 7d66a8320..4550c7710 100644
--- a/core/drivers/stm32_tamp.c
+++ b/core/drivers/stm32_tamp.c
@@ -22,6 +22,9 @@
 #include <mm/core_memprot.h>
 #include <sm/psci.h>
 #include <stdbool.h>
+#ifdef CFG_STM32MP25
+#include <stm32_sysconf.h>
+#endif /* CFG_STM32MP25 */
 #include <stm32_util.h>
 
 
@@ -1894,6 +1897,9 @@ static TEE_Result stm32_tamp_parse_fdt(struct stm32_tamp_platdata *pdata,
 	if (fdt_getprop(fdt, node, "wakeup-source", NULL))
 		pdata->is_wakeup_source = true;
 
+	if (fdt_getprop(fdt, node, "st,tamp-mask-pot-reset", NULL))
+		pdata->mask_pot_reset = true;
+
 	if (pdata->compat->tags & TAMP_HAS_RIF_SUPPORT) {
 		const fdt32_t *cuint = NULL;
 		uint32_t rif_conf = 0;
diff --git a/core/include/drivers/stm32_tamp.h b/core/include/drivers/stm32_tamp.h
index 0991d01eb..61aa7f5e5 100644
--- a/core/include/drivers/stm32_tamp.h
+++ b/core/include/drivers/stm32_tamp.h
@@ -125,6 +125,7 @@ struct stm32_tamp_platdata {
 	uint32_t pins_conf;
 	uint32_t out_pins;
 	bool is_wakeup_source;
+	bool mask_pot_reset;
 };
 
 #endif /* __DRIVERS_STM32_TAMP_H__ */
-- 
2.47.2

